### PR TITLE
[CELEBORN-432][FLINK] Refine RemoteShuffleOutputGate and RemoteShuffleInputGateFactory

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
Refine code in RemoteShuffleOutputGate

### Why are the changes needed?
ditto

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
UT

### CODE REVIEW

**Comment by waitinfuture** [comment]
ping @zhongqiangczq @RexXiong @FMX

File: client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java [hunk 1]
```diff
@@ -104,11 +104,9 @@ public RemoteShuffleOutputGate(
     this.mapId = shuffleDesc.getShuffleResource().getMapPartitionShuffleDescriptor().getMapId();
     this.attemptId =
         shuffleDesc.getShuffleResource().getMapPartitionShuffleDescriptor().getAttemptId();
-    this.rssMetaServiceHost =
-        ((RemoteShuffleResource) shuffleDesc.getShuffleResource()).getRssMetaServiceHost();
-    this.rssMetaServicePort =
-        ((RemoteShuffleResource) shuffleDesc.getShuffleResource()).getRssMetaServicePort();
-    this.shuffleWriteClient = createWriteClient();
+    this.rssMetaServiceHost = (shuffleDesc.getShuffleResource()).getRssMetaServiceHost();
```
Comment (RexXiong) [inline client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java:107]: We can remove the outer brackets at the same time, and I also suggest we can make the method call shorter than before.

**Review by zhongqiangczq — APPROVED** [review]

File: client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java (hunk 1 above)
Comment (waitinfuture) [reply client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java]: Brackets removed. What do you mean by "make the method call shorter than before" ?

Comment (RexXiong) [reply client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java]: like
shuffleDesc.getShuffleResource().getRssMetaServiceHost(); -> shuffleResource.getRssMetaServiceHost();
shuffleDesc.getShuffleResource().getRssMetaServicePort(); -> shuffleResource.getRssMetaServicePort();

But this is just a suggestion.

**Review by RexXiong — APPROVED** [review]

File: client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java (hunk 1 above)
Comment (waitinfuture) [reply client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java]: Yeah, I also feels that RemoteShuffleDescriptor nests too much, i.e. RemoteShuffleDescriptor.RemoteShuffleResource.ShuffleResourceDescriptor, maybe we can refactor later

**Pull request merged**